### ISSUE TITLE
How is the update for "train_on_batch" determined?

### ISSUE THREAD

**Issue by gxlzj** [fchollet/keras#3303]
Dear developers and friends,

I am a new user to keras. My question is as fellows:

I see for "train_on_batch", the optimizer only do one single gradient update. But how is the learning rate/ momentum used in this setting? 

For example, for SGD, we have a learning rate and momentum. When I use "train_on_batch" manually, will optimizer SGD automatically store the gradient update and use it as momentum when I use "train_on_batch"  again next time?

**Comment by codekansas** [comment]
Yes, it will, consult [here](https://github.com/fchollet/keras/blob/master/keras/engine/training.py#L686). Momentum is persistent between batches.
